source/ddbus/thin.d

```diff
@@ -18,26 +18,77 @@ public import ddbus.exception : wrapErrors, DBusException;
 struct ObjectPath {
   private string _value;
 
-  this(string objPath) {
-    enforce(_isValid(objPath));
+  this(string objPath) pure @safe {
+    enforce(isValid(objPath));
     _value = objPath;
   }
 
   string toString() const {
     return _value;
   }
 
-  size_t toHash() const pure nothrow @trusted {
+  /// Returns the string representation of this ObjectPath.
+  string value() const pure @nogc nothrow @safe {
```

Comment:
BTW, I just realized one of my upcoming PRs will actually allow to write: `obj.to!string` (variable `obj` being of type `ObjectPath`), which does exactly the same as `obj.value` here...

Reply:
the std.conv `to` does that too but I'm always so afraid of using `toString` methods because they can easily change. Also `value` is `pure @nogc nothrow @safe`, not sure if it is a good idea to add that to `toString` if it ever changes to returning something like `"ObjectPath " ~ value`.

After all, `toString` is supposed to be a human readable representation of the object: https://dlang.org/phobos/object.html#.Object.toString

Reply:
> After all, `toString` is supposed to be a human readable representation of the object

Which is what makes it different from `to!string` (though just conceptually in this case)

Reply:
but if you have a normal ObjectPath function you don't have the to function anyway, it is more likely that you then have std.conv:to which calls toString. I'm not saying to remove the custom `to!string` function but also having an explicit value for this sounds like a good idea to me. One could argue about which one gives cleaner code style, both work imo.

Reply:
Which one is cleaner is quite a matter of coding style. Let's keep `value` at least.

You are right about `std.conv.to` calling `toString`. I'll test if it also handles e.g. `to!dstring`, if it does, there is no reason to keep a custom `to` now. Though in that case I will put a comment in `toString` to prevent troubles arising from someone changing that one to be more descriptive without defining `to`.